app/src/main/java/model/place/Category.kt

```diff
@@ -17,48 +17,47 @@
 package com.guerinet.mymartlet.model.place
 
 import android.content.Context
-import androidx.room.Entity
-import androidx.room.PrimaryKey
+import com.google.firebase.firestore.DocumentSnapshot
 import com.guerinet.mymartlet.R
 import java.util.Locale
 
 /**
  * A type of place that the user can filter by
  * @author Julien Guerinet
  * @since 1.0.0
- *
- * @property id Category Id
- * @property en Category name in English
- * @property fr Category name in French
  */
-@Entity
-data class Category(
-    @PrimaryKey var id: Int = 0,
-    var en: String,
-    var fr: String
-) {
+class Category() {
 
-    /**
-     * Constructor used to create the Favorites (is [isFavorites] is true) and All types
-     */
-    constructor(isFavorites: Boolean) : this(if (isFavorites) FAVORITES else ALL, "", "")
+    var id: Int = ALL
+
+    var en: String = ""
+
+    var fr: String = ""
+
+    /** Localized category name */
+    val name: String
+        get() = if (Locale.getDefault().language == "fr") fr else en
 
     /**
-     * Returns the String to use. Uses the app [context]
+     * Constructor used to create the All category. Uses the app [context] to retrieve the necessary String
      */
-    fun getString(context: Context): String = when {
-        id == FAVORITES -> context.getString(R.string.map_favorites)
-        id == ALL -> context.getString(R.string.map_all)
-        Locale.getDefault().language == "fr" -> fr
-        else -> en
+    constructor(context: Context) : this() {
+        id = ALL
+        // Set the same translation for both languages, as this get regenerated every time the map is opened anyway
+        en = context.getString(R.string.map_all)
+        fr = en
     }
 
     companion object {
 
-        /** User's saved favorite places */
-        const val FAVORITES = -2
-
         /** All of the places */
         internal const val ALL = -1
+
+        /**
+         * Converts a Firestore [document] into a [Category] (null if error during the parsing)
+         */
+        fun fromDocument(document: DocumentSnapshot): Category? = document.toObject(Category::class.java)?.apply {
+            id = document.id.toInt()
```

Comment:
The `id` doesn't get mapped automatically so I map it manually after the object has been parsed by Firestore.